### ISSUE TITLE
Trying to replicate CIFAR-10 results

### ISSUE THREAD

**Issue by scfrank** [layer6ai-labs/lfr#1]
I am trying to replicate your CIFAR-10 results but am only reaching accuracy of 0.55, not 0.64 as in Table 11.
Can you help me figure out what's going on? Thanks!

I have added CIFAR to the code by copying the Kvasir encoders and target architectures.

In `ssl_models/models/targets.py` for the CIFAR10CNNTarget, the last nn.Linear layer is `nn.Linear(512, dim)` instead of `nn.Linear(3840, dim)`, to deal with the CIFAR image sizes (which I've kept at 32x32). Did you do something different here?

I am training with the following parameters: again following the kvasir script and modifying according to the paper. I may have missed something though.
(Also,  note `--target_layers 2 ` does nothing as far as I can tell, kvasir/cifar always have 3 layers; see `model_helper.build_target()`)

```
#!/usr/bin/env bash
epochs=400
eval_epochs=100
num_projectors=40

dim=2048
pred_dim=256
bs=512

python main.py --method lfr --dataset cifar10 -a resnet18\
        --num_targets $num_projectors  --target_layers 2\
        --init-beta --random-dropout\
        --target_sample_ratio 1 --num_of_classes 10\
        --epochs $epochs -b $bs --lr 0.03\
        --optimizer-type sgd --momentum 0.9 --wd 5e-4\
        --fix_pred_lr --train-predictor-individually\
        --pred_epochs 5 --pred_layers 2\
        --eval_epochs $eval_epochs --eval_lr 0.2 --eval_bs $bs\
        --dim $dim --pred_dim $pred_dim --loss barlow-batch\
        --num-of-runs 1 --workers 40\
        --save_freq 10 --eval_freq 10
```

**Comment by echoyi** [comment]
Hi, thanks for rasing this issue! 
For the CIFAR10 cnn target, can you try:

```
class CIFARCnnTarget(nn.Module):
    def __init__(self, dim, layers=2):
        super().__init__()
        self.dim = dim
        self.layers = layers
        self.network = nn.Sequential(
            nn.Conv2d(3, 8, kernel_size=3, padding=1),
            nn.ReLU(),
            nn.Conv2d(8, 16, kernel_size=3, stride=1, padding=1),
            nn.ReLU(),
            nn.MaxPool2d(2, 2), # output: 16 x 16 x 16

            nn.Conv2d(16, 32, kernel_size=3, stride=1, padding=1),
            nn.ReLU(),
            nn.Conv2d(32, 32, kernel_size=3, stride=1, padding=1),
            nn.ReLU(),
            nn.MaxPool2d(2, 2), # output: 32 x 8 x 8

            nn.Flatten(), 
            nn.Linear(32*8*8, dim))

    def forward(self, xb):
        return self.network(xb) 
```

Also thanks for pointing out the target_layer variable, it was a left over param from target architecture exploration but in the final experiments we ended up using the 4-layer cnns for both Kvasir and CIFAR, will fix it soon.

**Comment by scfrank** [comment]
Hi, thanks for your response! I've had some issues with my compute but I've been able to run with the new CnnTarget and unfortunately it doesn't fix the problem - I'm still getting test accuracies of maximum ~0.58.
- Could you share the script for running cifar-10, so I can be sure I have the correct settings?

In the course of trying to figure this out, I also discovered that I can't replicate the _supervised_ results for CIFAR-10 either: I get test accuracy of ~.77.
I ran `scripts/kvasir/run_kvasir_supervised.sh` and got final test accuracy of .727 (cf 83.2 in Table 2).
My guess is that this is due to the augmentations not being added. I can see 
`KvasirTransform`  but I don't see it being applied anywhere (e.g. in the dataloaders).
`get_transforms` in `utils/data_transforms.py` doesn't do anything with the image datasets (kvasir/cifar). 

I know what academic code is like! Thanks for helping me with this. (And if you'd rather I open a second issue for the augmentations, I can do that.)
